documentation/adr/0024-page-to-single-file.md

```diff
@@ -0,0 +1,47 @@
+[<-previous](0023-error-format.md)
+next->
+
+# 24 - Page PixelData as single file
+
+Date: 06-10-2022
+
+## Status
+
+Accepted
+
+## Context
+
+When performing file-backed operations, the [current implementation](0018-algorithm-migration-to-paged-files.md) of
+`PixelData` uses multiple files to store the temporary files and subsequently rebuilds them upon completion of the
+operation. While this might be efficient for partial (applied to subsections of data) access patterns, this is not the
+case for either full-sequential (applied to whole data [unary/binary ops]) or random-access (applied with variable
+stride across broad sections of data [cut]). In the Horace code, all operations are performed in either of these two
+modes, as such it is disadvantageous to use multiple temporary files, and instead a single file may be better.
+
+The reasoning behind a single file being better lies in several factors:
+
+- File system handling of multiple requests for opening can lead to slowdown as system looks up indices. These requests
+  come for every request to open a new file, which will happen multiple times with a series of files.
+- Multiple files may not be sequential on disc leading to inefficient long seek times as multiple processors attempt to
```

Comment:
"sequential" - do you mean "contiguous"?

Reply:
They might be contiguous but disordered, hence sequential, regardless the contiguity is irrelevant if they are separate files and the disc has to seek to find them in the index-table